src/ir/var.rs

```diff
@@ -258,12 +307,71 @@ impl ClangSubItemParser for Var {
                         let kind = ctx
                             .parse_callbacks()
                             .and_then(|c| c.int_macro(&name, value))
+                            .or_else(|| {
+                                default_limit_constant_type(&name, value)
+                            })
                             .unwrap_or_else(|| {
                                 default_macro_constant_type(ctx, value)
                             });
 
                         (TypeKind::Int(kind), VarType::Int(value))
                     }
+                    EvalResult::Cast(c, v) => {
+                        let ty = c
+                            .iter()
+                            .map(|v| std::str::from_utf8(v).unwrap())
+                            .collect::<Vec<&str>>();
+
+                        if let EvalResult::Int(Wrapping(value)) = *v {
+                            let val = VarType::Int(value);
+
+                            let kind = match &ty[..] {
+                                ["bool"] => IntKind::Bool,
+                                ["char"] => IntKind::Char {
+                                    is_signed: value < 0,
```

Comment:
Seems very weird to do this based on the value, can you elaborate? `char` is signed or unsigned depending on the platform.

Reply:
Is there a way to go from `"char"` to the corresponding `IntKind`? I guess this is a similar problem to the one with `wrapper_ids` above.